witchcraft-server/src/debug/mod.rs

```diff
@@ -42,32 +43,38 @@ pub trait Diagnostic {
 }
 
 pub struct DiagnosticRegistry {
-    diagnostics: HashMap<String, Arc<dyn Diagnostic + Sync + Send>>,
+    diagnostics: RwLock<HashMap<String, Arc<dyn Diagnostic + Sync + Send>>>,
```

Comment:
RwLock's overkill here IMO, since the lock calls are so short. I'd just go with a Mutex like we do in the MetricRegistry: https://github.com/palantir/witchcraft-rust-logging/blob/master/witchcraft-metrics/src/registry.rs#L51

Reply:
Ah, okay, changed!